docs/gen_pages.py

```diff
@@ -0,0 +1,38 @@
+"""Generate the code reference pages."""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("gatorgrade").rglob("*.py")):  #
+    module_path = path.relative_to("gatorgrade").with_suffix("")  #
+    doc_path = path.relative_to("gatorgrade").with_suffix(".md")  #
+    full_doc_path = Path("reference", doc_path)  #
+
+    parts = list(module_path.parts)
+
+    if parts[-1] == "__init__":  #
+        parts = parts[:-1]
+    elif parts[-1] == "__main__":
+        continue
+
+    nav[parts] = doc_path.as_posix()
+
+    with mkdocs_gen_files.open(full_doc_path, "w") as fd:  #
+        identifier = "gatorgrade." + ".".join(parts)  #
+        print("::: " + identifier, file=fd)  #
+
+    mkdocs_gen_files.set_edit_path(full_doc_path, path)  #
+
+#Generate navigation
+with mkdocs_gen_files.open("reference/Summary.md", "w") as nav_file:
+    # Generate source pages nav
+    # for file in sorted(Path("docs/source").rglob("*.md")):
+    #     title = file.relative_to("docs/source").with_suffix("")
+    #     path = file.relative_to("docs/")
+    #     print(f" * [{title}]({path})", file=nav_file)
```

Comment:
What's the reason this is commented out? If it's no longer needed then it may be best to remove it.